go/host/enclave/guardian.go

```diff
@@ -199,6 +199,9 @@ func (g *Guardian) PromoteToActiveSequencer() error {
 		g.logger.Error("Unable to promote to active sequencer, already active")
 		return nil
 	}
+	if g.state.enclaveL2Head.Cmp(big.NewInt(0)) > 0 && !g.state.IsLive() {
```

Comment:
is there an failure mode here where the backup sequencer never catches up with the L2 due to whatever killed the first enclave (bad batch or something) and never gets promoted?

Reply:
Yeah that's something to consider. But that's gonna require human intervention, the idea is that any batch in the host's database (which the enclave needs to be considered 'live') is considered to have been broadcast to the whole network.

If we decide we need to rewind that batch somehow and basically say its transactions never happened we'd have to delete it from the host DB and do some gymnastics to keep the validators happy. Might even require certifying a new version of the enclave image which had the bad batch surgery baked into it.

We should discuss this sort of stuff in our red teaming/security/bricking discussions.